src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -192,38 +193,39 @@ protected void onStop() {
 				backButtonHandler);
 	}
 
-	@Override protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
-		String requestCodeLabel = RequestCode.requestCodeToString(requestCode);
+	@Override protected void onActivityResult(int requestCd, int resultCode, Intent intent) {
+		RequestCode requestCode = RequestCode.valueOf(requestCd).get();
```

Comment:
We should probably check the optional before just calling `get()` (since the code here before at least seems to consider the possibility of an invalid requestCd).  Could do something like: 

```.java
	@Override protected void onActivityResult(int requestCd, int resultCode, Intent intent) {
		Optional<RequestCode> requestCodeOpt = RequestCode.valueOf(requestCd);
		if(requestCodeOpt.isPresent()) {
			RequestCode requestCode = requestCodeOpt.get();
			try {
				trace(this, "onActivityResult() :: requestCode=%s, resultCode=%s", requestCode.name(), resultCode);

				switch (requestCode) {
					case GRAB_PHOTO_ACTIVITY:
						photoGrabber.process(requestCode, resultCode, intent);
						return;
					case GRAB_MRDT_PHOTO_ACTIVITY:
						processMrdtResult(requestCode, resultCode, intent);
						return;
					case DISCLOSURE_LOCATION_ACTIVITY:
						processLocationPermissionResult(resultCode);
						return;
					case CHT_EXTERNAL_APP_ACTIVITY:
						processChtExternalAppResult(resultCode, intent);
						return;
					default:
						trace(this, "onActivityResult() :: no handling for requestCode=%s", requestCode.name());
				}
			} catch(Exception ex) {
				String action = intent == null ? null : intent.getAction();
				warn(ex, "Problem handling intent %s (%s) with requestCode=%s & resultCode=%s",
					intent, action, requestCode.name(), resultCode);
			}
		} else {
			trace(this, "onActivityResult() :: no handling for requestCode=%s", requestCd);			
		}
	}
```

(Sadly this is not quite as clean as it should be since it seems like our version of the Android SDK does not support the Optional.ifPresentOrElse(...) method...)

Reply:
Thanks! Done